src/buildercore/cfngen.py

```diff
@@ -81,53 +74,37 @@ def from_existing_context(field, default_value):
         'elasticache': False,
     }
 
+    # this is the context data from the currently existing template (if any)
+    # by re-using current values we can avoid making unnecessary changes when
+    # regenerating templates (like random passwords)
+    existing_context = more_context.pop('existing_context', {})
+
     context = copy.deepcopy(defaults)
     context.update(more_context)
 
-    assert context['stackname'] is not None, "a stackname wasn't provided."
+    # proceed with wrangling
+
+    ensure(context['stackname'], "a stackname wasn't provided.")
     stackname = context['stackname']
 
     # stackname data
-    bit_keys = ['project_name', 'instance_id', 'cluster_id']
-    bits = dict(zip(bit_keys, core.parse_stackname(stackname, all_bits=True)))
-    assert bits['project_name'] == pname, \
-        "the project name derived from the `stackname` doesn't match the given project name"
+    bits = core.parse_stackname(stackname, all_bits=True, idx=True)
+    ensure(bits['project_name'] == pname,
+           "the project name %r derived from the given `stackname` %r doesn't match" % (bits['project_name'], pname))
     context.update(bits)
 
+    # is this a production instance? if yes, then we'll do things like tweak the dns records ...
```

Comment:
do we? The only usage I could find is https://github.com/elifesciences/builder-base-formula/blob/29588e1f92b461be39eab77fbf7e6b7241d8a856/elife/postfix.sls#L8 which we could drop in favor of `pillar.elife.env == 'prod'`. Reducing the size of the context can make it easier to observe and debug

Reply:
oh wow - that's some primordial Salt code :) I'll fix that formula up and remove this key.

Reply:
pr: https://github.com/elifesciences/builder-base-formula/pull/99